packages/tests/tests/utils.spec.ts

```diff
@@ -115,3 +119,97 @@ describe("Util: toAsyncIterator: Filter", () => {
     expect(result.done).to.eq(true);
   });
 });
+
+// these tests are skipped until we can figure out how to mock the standalone functions
+// sinon doesn't seem to work with the standalone functions
```

Comment:
Make the code easier to test by splitting it in smaller functions  and splitting data retrieval from filtering logic.